internal/cloud/realm/user_test.go

```diff
@@ -140,3 +142,216 @@ func TestRealmUsers(t *testing.T) {
 		})
 	})
 }
+
+func TestProviderTypeIsValid(t *testing.T) {
+	for _, tc := range realm.ValidProviderTypes {
+		t.Run(fmt.Sprintf("%s should be valid", tc), func(t *testing.T) {
+			assert.Nil(t, realm.ProviderType(tc).IsValid())
+		})
+	}
+	t.Run(fmt.Sprintf("%s should be invalid", "invalid type"), func(t *testing.T) {
+		assert.Equal(t, realm.ProviderType("invalid type").IsValid(), errors.New("Invalid ProviderType"))
+	})
+}
+
+func TestProviderTypeDisplay(t *testing.T) {
+	for _, tc := range []struct {
+		pt             realm.ProviderType
+		expectedOutput string
+	}{
+		{
```

Comment:
for the relatively simple test case struct here, could you clean up some line heft here and leave these defined inline? 

```go
  {realm.AuthProviderTypeAnonymou, "Anonymous"},
  {realm.AuthProviderTypeUserPassword, "User/Password"},
  // etc
```